unlock-app/src/unlock.ts

```diff
@@ -9,11 +9,44 @@ export enum TransactionType {
 }
 
 export interface Transaction {
-  status: string,
-  confirmations: number,
-  createdAt: number,
-  hash: string,
-  lock: string,
-  name: string,
-  type: TransactionType,
+  status: string
+  confirmations: number
+  hash: string
+  lock: string
+  name: string
+  type: TransactionType
+  blockNumber: number
+  
+  key?: string // TODO: tighten up our types, hopefully we won't have too many
+               // optional properties.
+}
+
+// Analogous to the transactions in Redux
+export interface Transactions {
+  [hash: string]: Transaction
+}
+
+export interface Account {
+  address: string
+  balance: string
+}
+
+export interface Network {
+  name: number
+}
+
+// TransactionMetadata and TransactionMetadataMap are used in the log page to
+// pass additional information about transactions that do not belong in the
+// state.
+export interface TransactionMetadata {
```

Comment:
Indeed, this is weird. Could you add a TODO to look into doing something better here at some point?